Services/UserService/Commands/EditUserCommand.cs

```diff
@@ -29,11 +37,52 @@ public Task<bool> Execute(EditUserRequest request)
                     || string.IsNullOrEmpty(passwordRequest.NewPassword)))
             {
                 passwordChangeValidator.ValidateAndThrow(passwordRequest);
+            } else
+            {
+                throw new BadRequestException("Password is null or empty");
             }
             userInfoValidator.ValidateAndThrow(userInfo);
 
-            // TODO Realise edit user logic
-            throw new NotImplementedException();
+            string oldPasswordHash = ShaHash.GetPasswordHash(request.PasswordRequest.OldPassword);
```

Comment:
You have `var passwordRequest = request.PasswordRequest;` already, why `request.PasswordRequest.OldPassword` again

Reply:
Fixed